File: frontend/src/services/inventory/supplier.service.ts
import inventoryAPI from "./api";
import { ListSuppliersParams, SupplierInput } from "@/types/inventory/supplier";

export const SupplierService = {
  getAll: (params: ListSuppliersParams) => inventoryAPI.get("/suppliers/", { params }),
  getById: (id: string) => inventoryAPI.get(`/suppliers/${id}/`),
  create: (data: SupplierInput) => inventoryAPI.post("/suppliers/", data),
  update: (id: string, data: Partial<SupplierInput>) => inventoryAPI.patch(`/suppliers/${id}/`, data),
  remove: (id: string) => inventoryAPI.delete(`/suppliers/${id}/`),
};

File: inventory/services/supplier_service.py
from django.db import transaction

from inventory.models import Supplier


class SupplierService:

    @staticmethod
    @transaction.atomic
    def create(serializer):
        return serializer.save()

    @staticmethod
    @transaction.atomic
    def update(serializer):
        return serializer.save()

    @staticmethod
    @transaction.atomic
    def delete(supplier: Supplier):
        supplier.delete()